spec/index.md

```diff
@@ -3689,7 +3690,7 @@ Available = S.call_contexts[M.call_contexts].available_cycles
 )
 or
 ( M.entry_point = PublicMethod Name Caller Arg
-  F = query_as_update(Mod.query_methods[Name], Arg, Caller, Env)
+  F = query_as_update(Mod.query_methods[Name], Arg, Caller, Env, Available)
```

Comment:
We also need to change this:
```
    query_as_update(f, arg, caller, env, available) = λ wasm_state →
      match f(arg, caller, env, available)(wasm_state) with
```
the missing "caller" is a bug from before this PR.

Reply:
Done, although I'm not sure I understand the notation really. It feels like something is missing after "with".

Reply:
Oh, sorry for the confusion, we need to perform the change at the place where the function `query_as_update` is defined (further below), not at this line where I attached the comment (I can't open a comment on a line that is far from something that is changed by this PR).

Reply:
the change must happen a few lines above your change of `cycles_accepted = res.cycles_accepted;`

Reply:
Oh, of course, this makes sense now and I see what you mean about the missing "caller".